lib/src/config/route/app_routes.dart

```diff
@@ -12,13 +20,48 @@ abstract class AppRoutes {
         final MovieDetailsArguments arguments = ModalRoute.of(context)!
             .settings
             .arguments! as MovieDetailsArguments;
-        return MovieDetails(
-          movie: arguments.movie,
-          movieBloc: arguments.bloc,
-          castBloc: CastBloc(),
+        return Provider<MovieDetailsBloc>(
+          create: (_) => MovieDetailsBloc(
+            loadCastUseCase: LoadCastUseCase(
+              repository: APIRepository(client: Client()),
+              database: Provider.of<MovieDatabase>(
+                context,
+                listen: false,
+              ),
+            ),
+          ),
+          builder: (BuildContext builderContext, _) {
+            return MovieDetails(
+              movie: arguments.movie,
+              movieBloc: Provider.of<MovieDetailsBloc>(
+                builderContext,
+                listen: false,
+              ),
+              setLastMovie: arguments.setLastMovie,
+            );
+          },
         );
       },
-      Routes.homeRouteName: (BuildContext context) => HomePage(),
+      Routes.homeRouteName: (BuildContext context) => Provider<MoviesBloc>(
+            create: (BuildContext ctx) => MoviesBloc(
+              getMoviesUseCase: MoviesByTypeUseCase(
+                repository: APIRepository(client: Client()),
+                database: Provider.of<MovieDatabase>(
+                  ctx,
+                  listen: false,
+                ),
+              ),
+              fetchFirstMovie: FetchLastSeenMovieUseCase(),
+            ),
+            builder: (BuildContext builderContext, _) {
```

Comment:
Use the same parameter name to BuildContext, currently you are using (context, ctx and builderContext)

Reply:
Done